### ISSUE TITLE
Instructions on training

### ISSUE THREAD

**Issue by zhengpingzhou** [fwtan/who_where#2]
Thanks for your insightful work and helpful codes! the idea is really inspiring and the codes are nicely organized.

I'm trying to train my own data on your bbox-prediction network (in `PredictionCNN.py`). There're 2 main issues about it:
1. Would you please kindly provide an instruction on training? (i.e. data formats, caveats, training time, etc.)
2. Variable `cfg.TOP_K` in function `generator` is undefined. I could not find any explanation about it either in the paper or in other parts of the codes.

Thanks in advance!

**Comment by fwtan** [comment]
I will try to provide a training script (hopefully soon...) 
Basically, you may need to provide the inpainted image, instead of the original color image, as the input for training. We include an [Adobe script](https://github.com/fwtan/who_where/blob/master/tools/inpainting.js) to automatically in-paint the COCO images via Photoshop. 
The inpainted images should be placed in the COCO data directory, as indicated in the inpainted_path_from_indices function in [coco.py](https://github.com/fwtan/who_where/blob/master/lib/datasets/coco.py).
The cfg.TOP_K is a parameter controlling the number of locations and shapes of the bounding boxes sampled from the model. Like in the paper, we evaluated our top-8 results, which are produced from the combinations of the top 2 location predictions, top 2 size
predictions and top 2 retrieved segments.

**Comment by Paula15** [comment]
Thanks for your response! I finally successfully ran training by ignoring that variable and removing the `np.repeat...` statements in `generator`.

Besides, as mentioned in your answer, inpainting is done by Adobe Photoshop. Is there any reason for not using open source implementations such as [context encoder](https://github.com/pathak22/context-encoder) or [generative inpainting](https://github.com/JiahuiYu/generative_inpainting)?

**Comment by fwtan** [comment]
We choose to use Photoshop as it can handle masks with arbitrary shapes and achieve reasonable visual quality. 
But feel free to try other methods such as [Globally and Locally Consistent Image Completion](http://hi.cs.waseda.ac.jp/~iizuka/projects/completion/en/) and [generative inpainting](https://github.com/JiahuiYu/generative_inpainting). 
There is of course NO reason for not using these open source methods if they can help achieve better results!
